src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -32,6 +33,27 @@ static void Main()
         }
 
         [Fact]
+        [WorkItem(19013, "https://github.com/dotnet/roslyn/issues/19013")]
+        public void TestCSharp7Cascade()
+        {
+            string source = @"
+using System.Threading;
+using System.Threading.Tasks;
+
+class C
+{
+    async Task M(CancellationToken t = default) { await Task.Delay(0); }
+}
+";
+            var comp = CreateCompilationWithMscorlib46(source);
+            comp.VerifyDiagnostics(
+                // (7,40): error CS8107: Feature 'default literal' is not available in C# 7. Please use language version 7.1 or greater.
+                //     async Task M(CancellationToken t = default) { await Task.Delay(0); }
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7, "default").WithArguments("default literal", "7.1").WithLocation(7, 40)
```

Comment:
@AlekseyTs When checking value on "default" (which has a parser error), it gets transformed to a bad expression by the code below.

```C#
        internal bool CheckValueKind(BoundExpression expr, BindValueKind valueKind, DiagnosticBag diagnostics)
        {
            if (expr.HasAnyErrors)
            {
                return false;
            }
            ...
        }

        private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind, DiagnosticBag diagnostics)
        {
            ...
            if (!hasResolutionErrors && CheckValueKind(expr, valueKind, diagnostics) ||
                expr.HasAnyErrors && valueKind == BindValueKind.RValueOrMethodGroup)
            {
                return expr;
            }

            var resultKind = (valueKind == BindValueKind.RValue || valueKind == BindValueKind.RValueOrMethodGroup) ?
                LookupResultKind.NotAValue :
                LookupResultKind.NotAVariable;

            return ToBadExpression(expr, resultKind);
        }
```